### PR TITLE
Candidate 0.1.1

### PR DESCRIPTION
**Pull request by pde**
0.1.1 has been released; this branch contains its tag (as well as finalised release scripts)

### CODE REVIEW

File: tools/release.sh [hunk 1]
```diff
@@ -97,16 +135,44 @@ pip install \
...
+done
+deactivate
+
+cd ..
+echo Now in $PWD
+name=${root_without_le%.*}
+ext="${root_without_le##*.}"
+rev="$(git rev-parse --short HEAD)"
+echo tar cJvf $name.$rev.tar.xz $name.$rev
+echo gpg -U $RELEASE_GPG_KEY --detach-sign --armor $name.$rev.tar.xz
```
Comment (bmw) [inline tools/release.sh]: These two commands are the ones that are automated by `dev-release2.sh`, but not this script.

File: tools/release.sh [hunk 2]
```diff
@@ -1,13 +1,43 @@
-#!/bin/sh -xe
+#!/bin/bash -xe
 # Release dev packages to PyPI
 
+Usage() {
+    echo Usage:
+    echo "$0 [ --production ]"
```
Comment (bmw) [inline tools/release.sh]: I think we should explain what that if `--production` is included, you must also specify the current and next versions, but we obviously don't have to block this PR on that.

**Comment by bmw** [comment]
This also doesn't include the tag (see [this](https://stackoverflow.com/questions/12278660/adding-tags-to-a-pull-request)).

Also, is this supposed to include the version bump or is that in another PR?

**Comment by pde** [comment]
I accidentally pushed this branch from a working checkout, rather than the `releases/le-0.1.1.XX/` subdirectory.  Correcting that now gets us the signed release and version bump; I've also pushed the tag.

**Comment by bmw** [comment]
Looking at the changes, it looks like a version bump wasn't even necessary. The version was already set at `0.2.0.dev0`.

It also looks like the changes made to the version number by the release scripts are incorrect.

**Comment by kuba** [comment]
I would like to note that "this branch contains its tag" statement doesn't make much sense. Both branches and tags are git refs. You could say that this branch contains a commit happens to be tagged, but pulling in the branch doesn't mean that the tag is pulled in. They are separate things!

File: acme/setup.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 from setuptools import find_packages
 
 
-version = '0.2.0.dev0'
+version = '0.2.0dev0'
```
Comment (kuba) [inline acme/setup.py]: why do you remove the dot? `version` has been kept so far in canonical form, with dot

File: letshelp-letsencrypt/setup.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
 from setuptools import find_packages
 
 
-version = '0.2.0.dev0'
+version = '0.2.0'
```
Comment (kuba) [inline letshelp-letsencrypt/setup.py]: that's wrong!

File: tools/half-sign.c [hunk 5]
```diff
@@ -0,0 +1,117 @@
+#include <unistd.h>
```
Comment (kuba) [inline tools/half-sign.c]: oh wow, when did we start using C? Where is a doc for this file? what does it do?

File: tools/release.sh [hunk 6]
```diff
@@ -1,13 +1,43 @@
-#!/bin/sh -xe
+#!/bin/bash -xe
```
Comment (kuba) [inline tools/release.sh]: why change from POSIX to bash-only?

Comment (pde) [inline tools/release.sh]: To allow the use of bash string editing features to keep our release script as sane as possible.

File: acme/setup.py (hunk 3 above)
Comment (pde) [inline acme/setup.py]: Thanks for catching that.  Human error!

File: tools/half-sign.c (hunk 5 above)
Comment (pde) [inline tools/half-sign.c]: Explanation added. This isn't in use yet, but we're likely to add airgapped signatures from that code in the future.

**Comment by bmw** [comment]
LGTM

File: tools/half-sign.c (hunk 5 above)
Comment (kuba) [inline tools/half-sign.c]: if it's not used anywhere then remove it...

**Pull request merged**